### PR TITLE
feat: add java api and test

### PR DESCRIPTION
**Pull request by zong-zhe**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [x] N
- [ ] Y 

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [x] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Other

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [x] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [x] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [ ] Other

### CODE REVIEW

File: java/src/test/java/com/kcl/ListVariablesTest.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+                    .setFile("./src/test_data/list_variables/main.k").addSpecs(spec).build());
+
+            Assert.assertEquals(result.getVariablesCount(), 1);
+            Assert.assertEquals(result.getVariablesMap().get(spec).getValue(), expectedValue);
+
+        }
+    }
+
+    @Test
+    public void testProgramSymbolsWithCache() throws Exception {
```
Comment (Peefy) [inline java/src/test/java/com/kcl/ListVariablesTest.java:47]: Remove dup test cases.

File: java/src/test/java/com/kcl/OverrideFileTest.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+            Assert.assertEquals(result.getResult(), true);
+        }
+        if (Files.exists(target)) {
+            Files.delete(target);
+        }
+    }
+
+    @Test
+    public void testProgramSymbolsWithCache() throws Exception {
```
Comment (Peefy) [inline java/src/test/java/com/kcl/OverrideFileTest.java:53]: Remove dup test cases.

Comment (zong-zhe) [reply java/src/test/java/com/kcl/OverrideFileTest.java]: fixed

File: java/src/test/java/com/kcl/ListVariablesTest.java (hunk 1 above)
Comment (zong-zhe) [reply java/src/test/java/com/kcl/ListVariablesTest.java]: fixed

**Review by Peefy — APPROVED** [review]
LGTM

**Pull request merged**